cache/manager.go

```diff
@@ -104,7 +104,7 @@ func (cm *cacheManager) get(ctx context.Context, id string, opts ...RefOption) (
 	defer rec.mu.Unlock()
 
 	if rec.mutable {
-		if len(rec.refs) != 0 {
+		if rec.dead || len(rec.refs) != 0 {
 			return nil, errors.Wrapf(errLocked, "%s is locked", id)
```

Comment:
`if rec.dead { return nil, errors.Errorf("%s is being finalized", id) }`?

Reply:
This error means that you can't obtain a mutable ref because something else is keeping an immutable one. "Finalizing" is just the process of freezing the immutable ref(because it becomes a parent for some other snapshot) so that a mutable one can never be returned. The extra "dead" state is here so we don't need to lock all snapshots while calling mount/finalize, so we just mark it as dead and actually remove it later.